File: python/1598.py
# Link: https://leetcode.com/problems/crawler-log-folder/description/
# Runtime: 55 ms
# Memory: 16.78 MB

class Solution:
    def minOperations(self, logs: List[str]) -> int:
        minimum_operatons_to_go_back = 0
        for log in logs:
            if log == '../':
                minimum_operatons_to_go_back = max(0, minimum_operatons_to_go_back - 1)
            elif log != './':
                minimum_operatons_to_go_back += 1
        
        return minimum_operatons_to_go_back

File: c++/1598.cpp
// Link: https://leetcode.com/problems/crawler-log-folder/description/
// Runtime: 0 ms
// Memory: 13.81 MB

class Solution {
public:
  int minOperations(const std::vector<std::string> &logs) {
    int minimumOperatonsToGoBack = 0;
    for (const auto &log : logs)
      if (log == "../")
        minimumOperatonsToGoBack = std::max(0, minimumOperatonsToGoBack - 1);
      else if (log != "./")
        minimumOperatonsToGoBack += 1;

    return minimumOperatonsToGoBack;
  }
};
